API.md

```diff
@@ -0,0 +1,440 @@
+| `logoIsHidden`  | `boolean` | Optional | Whether logo is visible in lower left corner. | `false` |
+| `compassIsHidden`  | `boolean` | Optional | Whether compass is visible when map is rotated. | `false` |
+| `contentInset` | `array` | Optional | Change the padding of the viewport of the map. Offset is in pixels. `[top, right, bottom, left]` `[0, 0, 0, 0]` |
+| `style`  | React styles | Optional | Styles the actual map view container | N/A |
+| `debugActive`  | `boolean` | Optional | Turns on debug mode. | `false` |
+
+## Callback props
+
+```javascript
+<MapView onSomethingHappened={payload => {
+  //...
+}}/>
+```
+
+| Prop | Payload shape | Description
+|---|---|---|
+| `onRegionWillChange` | `{latitude: 0, longitude: 0, zoomLevel: 0, direction: 0, pitch: 0, animated: false}` | Fired when the map begins panning or zooming. `animated` indicates whether the action is user-driven or animation-driven.
+| `onRegionDidChange` | `{latitude: 0, longitude: 0, zoomLevel: 0, direction: 0, pitch: 0, animated: false}` | Fired when the map ends panning or zooming.
+| `onOpenAnnotation` | `{id: 'marker_id', title: null, subtitle: null, latitude: 0, longitude: 0}` | Fired when tapping an annotation.
+| `onRightAnnotationTapped` | `{id: 'marker_id', title: null, subtitle: null, latitude: 0, longitude: 0}` | Fired when user taps the `rightCalloutAccessory` of an annotation.
+| `onChangeUserTrackingMode` | `Mapbox.userTrackingMode.none` | Fired when the user tracking mode gets changed by an user pan or rotate.
+| `onUpdateUserLocation` | `{latitude: 0, longitude: 0, headingAccuracy: 0, magneticHeading: 0, trueHeading: 0, isUpdating: false}` | Fired when the user's location updates.
+| `onLocateUserFailed` | `{message: 'Error message'}` | Fired when there is an error getting the user's location. Do not rely on the string that is returned for determining what kind of error it is.
+| `onTap` | `{latitude: 0, longitude: 0, screenCoordX: 0, screenCoordY: 0}` | Fired when the users taps the screen.
+| `onLongPress` | `{latitude: 0, longitude: 0, screenCoordX: 0, screenCoordX: 0}` | Fired when the user taps and holds screen for 1 second.
+| `onStartLoadingMap` | `undefined` | Fired once the map begins loading the style. |
+| `onFinishLoadingMap` | `undefined` | Fired once the map has loaded the style. |
+
+## Methods
+
+You first need to get a ref to your `MapView` component:
+
+```jsx
+<MapView ref={map => { this._map = map; }} />
+```
+
+Then call methods as `this._map.methodName()`.
+
+---
+
+```javascript
+this._map.setDirection(direction, animated = true, callback);
+this._map.setZoomLevel(zoomLevel, animated = true, callback);
+this._map.setCenterCoordinate(latitude, longitude, animated = true, callback);
+this._map.setCenterCoordinateZoomLevel(latitude, longitude, zoomLevel, animated = true, callback);
+this._map.setCenterCoordinateZoomLevelPitch(latitude, longitude, zoomLevel, pitch, animated = true, callback); // Android only
```

Comment:
The equivalent on iOS would be to set the map camera. On either platform, when the map is tilted, the map’s “zoom level” doesn’t actually correspond to the zoom level of the tiles that are visible on the map. So MapKit and the Mapbox iOS SDK let you [create a camera](https://www.mapbox.com/ios-sdk/api/3.3.0-beta.3/Classes/MGLMapCamera.html) based on the altitude above the ground or the distance from the focal point.